misc/testerina/modules/testerina-core/src/main/ballerina/assert.bal

```diff
@@ -238,34 +238,35 @@ public isolated function assertFail(string msg = "Test Failed!") returns never {
-        string errorMsg = "";
-        string expectedStr = sprintf("%s", expected);
-        string actualStr = sprintf("%s", actual);
-        if (expectedStr.length() > maxArgLength) {
-            expectedStr = getFormattedString(expectedStr);
-        }
-        if (actualStr.length() > maxArgLength) {
-            actualStr = getFormattedString(actualStr);
-        }
-        if (expectedType != actualType) {
-            errorMsg = string `${msg}` + "\n \nexpected: " + string `<${expectedType}> '${expectedStr}'` + "\nactual\t: "
+isolated function getInequalityErrorMsg(any|error actual, any|error expected, string msg = "\nAssertion Failed!")
+        returns @tainted string {
+    string expectedType = getBallerinaType(expected);
+    string actualType = getBallerinaType(actual);
+    string errorMsg = "";
+    string expectedStr = sprintf("%s", expected);
+    string actualStr = sprintf("%s", actual);
+    if expectedStr.length() > maxArgLength {
+        expectedStr = getFormattedString(expectedStr);
+    }
+    if actualStr.length() > maxArgLength {
+        actualStr = getFormattedString(actualStr);
+    }
+    if expectedType != actualType {
+        errorMsg = string `${msg}` + "\n \nexpected: " + string `<${expectedType}> '${expectedStr}'` + "\nactual\t: "
                 + string `<${actualType}> '${actualStr}'`;
-        } else if (actual is string && expected is string) {
-            string diff = getStringDiff(<string>actual, <string>expected);
-            errorMsg = string `${msg}` + "\n \nexpected: " + string `'${expectedStr}'` + "\nactual\t: "
-                                     + string `'${actualStr}'` + "\n \nDiff\t:\n" + string `${diff}`;
-        } else if (actual is map<anydata> && expected is map<anydata>) {
-            string diff = getMapValueDiff(<map<anydata>>actual, <map<anydata>>expected);
-            errorMsg = string `${msg}` + "\n \nexpected: " + string `'${expectedStr}'` + "\nactual\t: " +
+    } else if actual is string && expected is string {
+        string diff = getStringDiff(<string>actual, <string>expected);
+        errorMsg = string `${msg}` + "\n \nexpected: " + string `'${expectedStr}'` + "\nactual\t: "
+                                    + string `'${actualStr}'` + "\n \nDiff\t:\n" + string `${diff}`;
+    } else if actual is map<anydata> && expected is map<anydata> {
+        string diff = getMapValueDiff(<map<anydata>>actual, <map<anydata>>expected);
+        errorMsg = string `${msg}` + "\n \nexpected: " + string `'${expectedStr}'` + "\nactual\t: " +
                             string `'${actualStr}'` + "\n \nDiff\t:\n" + string `${diff}`;
-        } else {
-            errorMsg = string `${msg}` + "\n \nexpected: " + string `'${expectedStr}'` + "\nactual\t: "
-                                                 + string `'${actualStr}'`;
-        }
-        return errorMsg;
+    } else {
+        errorMsg = string `${msg}` + "\n \nexpected: " + string `'${expectedStr}'` + "\nactual\t: "
```

Comment:
Can't we use a single string template

Reply:
Fixed with 27c521a5975fb3e867c8bdc46b9cc676e14f992c